src/H5Ppublic.h

```diff
@@ -8397,6 +8402,45 @@ H5_DLL herr_t H5Pget_selection_io(hid_t plist_id, H5D_selection_io_mode_t *selec
  */
 H5_DLL herr_t H5Pget_no_selection_io_cause(hid_t plist_id, uint32_t *no_selection_io_cause);
 
+/**
+ * \ingroup DXPL
+ *
+ * \brief Retrieves the type of I/O that HDF5 actually performed on
+ *        last I/O call (not necessarily the type requested)
+ *
+ * \dxpl_id{plist_id}
+ * \param[out] actual_selection_io_mode A bitwise set value indicating the
+ *                                      type of I/O performed
+ * \return \herr_t
+ *
+ * \par Motivation:
+ *      A user can request selection I/O to be performed via a data transfer
+ *      property list (DXPL).  This can be used to enable collective I/O with
+ *      type conversion, or with custom VFDs that support vector or selection
+ *      I/O.  However, there are conditions that can cause HDF5 to forgo
+ *      selection or vector I/O and perform legacy (scalar) I/O instead.
```

Comment:
<after existing text> This function allows the user to determine which type or types of I/O were actually performed.

Reply:
We should add a note stating: "Be aware that this function will include the types of all I/O that were performed during this operation, including any metadata operations that may be incidental to the requested I/O. To make sure this function is only capturing raw data I/O, one can disable metadata cache evictions by calling H5Fset_mdc_config() with evictions_enabled set to false."

Reply:
My intent with the original comment was to add that sentence to the end of the Motivation section. I'll go ahead and make my changes and paste the entire doxygen comment here

Reply:
> /**
 * \ingroup DXPL
 *
 * \brief Retrieves the type(s) of I/O that HDF5 actually performed on
 *        last I/O call (not necessarily the type requested)
 *
 * \dxpl_id{plist_id}
 * \param[out] actual_selection_io_mode A bitwise set value indicating the
 *                                      type(s) of I/O performed
 * \return \herr_t
 *
 * \par Motivation:
 *      A user can request selection I/O to be performed via a data transfer
 *      property list (DXPL).  This can be used to enable collective I/O with
 *      type conversion, or with custom VFDs that support vector or selection
 *      I/O.  However, there are conditions that can cause HDF5 to forgo
 *      selection or vector I/O and perform legacy (scalar) I/O instead.  This
 *      function allows the user to determine which type or types of I/O were
 *      actually performed.
 *
 * \details H5Pget_actual_selection_io_mode() allows the user to determine which
 *          type(s) of I/O were actually performed during the last I/O operation
 *          which used \p plist_id.  This property is set after all I/O is
 *          completed; if I/O fails, it will not be set.
 *
 *          H5Pget_no_selection_io_cause() can be used to determine the reason
 *          why selection or vector I/O was not performed.
 *
 *          Valid values returned in \p actual_selection_io_mode are listed
 *          as follows.
 *
 *          - #H5D_SCALAR_IO
 *          Scalar (or legacy MPIO) I/O was performed
 *          - #H5D_VECTOR_IO
 *          Vector I/O was performed
 *          - #H5D_SELECTION_IO
 *          Selection I/O was performed
 *
 *          Be aware that this function will include the types of all I/O that
 *          were performed during this operation, including any metadata
 *          operations that may be incidental to the requested I/O. To make sure
 *          this function is only capturing raw data I/O, one can temporarily
 *          disable metadata cache evictions by calling H5Fset_mdc_config() with
 *          evictions_enabled set to false.
 *
 * \since 1.14.2
 *
 */

Reply:
I can't get github to avoid parsing the doxygen (even in a code block) so I'll jsut email it to you.